2-With-broker/README.md

```diff
@@ -248,24 +262,55 @@ MSAL uses `canOpenURL:` to check if the broker is installed on the device. In
 
 #### Step 6: Register your RedirectUri in the application portal
 Using the broker adds an extra requirement on your redirectUri. The redirectUri _**must**_ have the following format:
+
+**For iOS:**
 ```CSharp
 $"msauth.{BundleId}://auth"
 ```
-**For example:**
+**Example for iOS:**
 ```CSharp
 public static string redirectUriOnIos = "msauth.com.yourcompany.XForms://auth"; 
 ```
-**You'll notice the RedirectUri matches the `CFBundleURLSchemes` name you included in the `Info.plist` file.**
+**You'll notice the iOS RedirectUri matches the `CFBundleURLSchemes` name you included in the `Info.plist` file.**
+
+**For Android:**
+```CSharp
+$"msauth://{Package Name}/{Signature Hash}"
+```
+**Example for Android:**
+```CSharp
+public static string redirectUriOnIAndroid = "msauth://UserDetailsClient.Droid/hgbUYHVBYUTvuvT&Y6tr554365466="; 
+```
+
+Once you have created your redirect URI, you can register it by navigating to the **Authentication** tab; under the **Redirect URIs** section, select **Public client/native (mobile & desktop)** entry from the dropdown box and paste in your redirect URI next to it. Click on save at the top of the page to save changes.
+
+**If you have the new experience**
+
+- **Android:** Select **Add a platform** under the **Platform Configurations** section, click on **Android** and add your package name and the signature hash.
 
-#### Step 7: make sure the redirect URI is registered with your app
+- **iOS:** Select **Add a platform** under the **Platform Configurations** section, click on **iOS** and add your iOS bundle id.
+
+Once you have the proper redirect URI, update the `BrokerRedirectUriOnIos` and the `BrokerRedirectUriOnAndroid` fields in the App.cs file with the values.
+
+#### Android Specific Redirect URI Configuration
+
+The redirect uri on Android will need to be created based on the signature of the .APK used to sign it. This means that it will be different depending on where this sample is run because Visual Studio creates a unique signing key for debugging purposes on every machine. You can figure out what that signature will be by running the following commands
+
+- For Windows: `keytool -exportcert -alias androiddebugkey -keystore %HOMEPATH%\.android\debug.keystore | openssl sha1 -binary | openssl base64`
+- For Mac: `keytool -exportcert -alias androiddebugkey -keystore ~/.android/debug.keystore | openssl sha1 -binary | openssl base64`
+
+More information on the [keytool here](https://docs.oracle.com/cd/E82085_01/140/rib_security_guide/appendixB.htm)
+
+Once you have your signature, simply use the `msauth://"Package Name"/"Signature Hash"` format as shown above to create your redirect URI.
+
+NOTE: You also have the option of acquiring your redirect URI with code. see [Brokered Authentication for Android](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-use-brokers-with-xamarin-apps) for more details
```

Comment:
What does this paragraph mean? The msdn page has no info on Android broker.

Reply:
this will be updated to match the MSAL wiki soon